src/Microsoft.AspNet.OData.Shared/Query/Expressions/SelectExpandWrapperJsonConverter.cs

```diff
@@ -0,0 +1,308 @@
+            }
+
+            private static JsonPropertyNameAttribute GetJsonPropertyNameAttribute(PropertyInfo property)
+            {
+                return property.GetCustomAttributes(typeof(JsonPropertyNameAttribute), inherit: false)
+                    .OfType<JsonPropertyNameAttribute>().SingleOrDefault();
+            }
+        }
+
+        /// <summary>
+        /// Represents a custom <see cref="JsonConverter"/> to serialize <see cref="SelectExpandBinder.SelectSome{TElement}"/> instances to JSON.
+        /// </summary>
+        private class SelectSomeOfTJsonConverter<TEntity>
+            : JsonConverter<SelectExpandBinder.SelectSome<TEntity>>
+        {
+            /// <inheritdoc/>
+            public override SelectExpandBinder.SelectSome<TEntity> Read(
+                ref Utf8JsonReader reader,
+                Type typeToConvert,
+                JsonSerializerOptions options)
+            {
+                throw new NotImplementedException(
+                    Error.Format(
+                        SRResources.JsonConverterDoesNotSupportRead,
+                        typeof(SelectExpandBinder.SelectSome<TEntity>).Name));
+            }
+
+            /// <inheritdoc/>
+            public override void Write(
+                Utf8JsonWriter writer,
+                SelectExpandBinder.SelectSome<TEntity> value,
+                JsonSerializerOptions options)
+            {
+                JsonSerializer.Serialize(
+                    writer,
+                    value.ToDictionary(_mapperProvider), typeof(IDictionary<string, object>),
+                    options);
+            }
+        }
+
+        /// <summary>
+        /// Represents a custom <see cref="JsonConverter"/> to serialize <see cref="SelectExpandBinder.SelectSomeAndInheritance{TElement}"/> instances to JSON.
+        /// </summary>
+        private class SelectSomeAndInheritanceOfTJsonConverter<TEntity>
+            : JsonConverter<SelectExpandBinder.SelectSomeAndInheritance<TEntity>>
+        {
+            /// <inheritdoc/>
+            public override SelectExpandBinder.SelectSomeAndInheritance<TEntity> Read(
+                ref Utf8JsonReader reader,
+                Type typeToConvert,
+                JsonSerializerOptions options)
+            {
+                throw new NotImplementedException(
+                    Error.Format(
+                        SRResources.JsonConverterDoesNotSupportRead,
+                        typeof(SelectExpandBinder.SelectSomeAndInheritance<TEntity>).Name));
+            }
+
+            /// <inheritdoc/>
+            public override void Write(
+                Utf8JsonWriter writer,
+                SelectExpandBinder.SelectSomeAndInheritance<TEntity> value,
+                JsonSerializerOptions options)
+            {
+                JsonSerializer.Serialize(
+                    writer,
+                    value.ToDictionary(_mapperProvider),
```

Comment:
Does the Newtonsoft converter version also use `ToDictionary`?

Reply:
@habbes Yes it does https://github.com/OData/WebApi/blob/b810e7185d45104f16fcd8ae1fc18cfeef923658/src/Microsoft.AspNet.OData.Shared/Query/Expressions/SelectExpandWrapperConverter.cs#L47

Reply:
I don't understand this answer..